File: python/Command/ReadFileCommand.py
from ICommand import ICommand
import FileHandler
import string

class ReadFileCommand(ICommand):
    def __init__(self, fileHandler: FileHandler, fileName: string) -> None:
        print("COMMAND: Creating command\n")
        self.fileHandler = fileHandler
        self.fileName = fileName

    def Execute(self) -> None:
        print("COMMAND: Executing command from invoker, delegating to logic\n")
        self.fileHandler.ReadFile(self.fileName)

File: C#/Command/ReadFileCommand.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Command
{
    class ReadFileCommand : ICommand
    {
        private FileHandler fileHandler;
        private string fileName;

        public ReadFileCommand(FileHandler fileHadnler, string fileName)
        {
            Console.WriteLine("COMMAND: Creating command");
            this.fileHandler = fileHadnler;
            this.fileName = fileName;
        }

        public void Execute()
        {
            Console.WriteLine("COMMAND: Executing command from invoker, delgating to logic");
            fileHandler.ReadFile(fileName);
        }
    }
}
